src/main/kotlin/io/inventi/eventstore/Subscriptions.kt

```diff
@@ -19,11 +19,11 @@ abstract class Subscriptions<T : EventstoreEventHandler>(private val handlers: L
         }
     }
 
-    protected abstract fun ensureSubscription(handler: T)
+    internal abstract fun ensureSubscription(handler: T)
 
-    protected abstract fun startSubscription(handler: T, onFailure: (FailureType) -> Unit)
+    internal abstract fun startSubscription(handler: T, onFailure: (FailureType) -> Unit)
 
-    protected open fun dropSubscription(handler: T) { }
+    internal abstract fun dropSubscription(handler: T)
```

Comment:
I feel like `startSubscription(handler)` and `dropSubscription(handler: T)` are implementation details and should not be exposed. We could keep them protected and instead have 2 functions `startSubscriptions()` and `dropSubscriptions()` which would be called when bean is initialized and destroyed  e.g.:

```
abstract class Subscriptions<T : EventstoreEventHandler>(private val handlers: List<T>) : InitializingBean, DisposableBean {
    protected val logger by LoggerDelegate()

    override fun afterPropertiesSet() {
        startSubscriptions()
    }

    override fun destroy() {
        dropSubscriptions()
    }

    internal open fun startSubscriptions() {
        handlers.forEach {
            ensureSubscription(it)
            startSubscription(it, onSubscriptionFailure(it))
        }
    }

    internal open fun dropSubscriptions() {
        handlers.forEach {
            dropSubscription(it)
        }
    }

    protected abstract fun ensureSubscription(handler: T)

    protected abstract fun startSubscription(handler: T, onFailure: (FailureType) -> Unit)

    prote
```

Reply:
Done